src/model/CoilCoolingDXMultiSpeedStageData.cpp

```diff
@@ -416,39 +417,39 @@ namespace detail {
     return t_clone;
   }
 
-  boost::optional<std::tuple<int, CoilCoolingDXMultiSpeed>> CoilCoolingDXMultiSpeedStageData_Impl::stageIndexAndParentCoil() const {
+  std::vector<IdfObject> CoilCoolingDXMultiSpeedStageData_Impl::remove() {
+    if (auto _coil = parentCoil()) {
+      _coil->removeStage(getObject<CoilCoolingDXMultiSpeedStageData>());
+    }
+    return ParentObject_Impl::remove();
+  }
```

Comment:
When remove a CoilCoolingDXMultiSpeedStageData, delete any extensible groups for the parentCoil if any